### ISSUE TITLE
The metrics application cannot be restarted

### ISSUE THREAD

**Issue by peergynt** [nlighten/tomcat_exporter#15]
If I want to restart the /metrics application (not tomcat, just the application), the application fails to start and I see the following exception:
```
java.lang.IllegalArgumentException: Collector already registered that provides name: tomcat_session_active_total
	io.prometheus.client.CollectorRegistry.register(CollectorRegistry.java:54)
	io.prometheus.client.Collector.register(Collector.java:139)
	io.prometheus.client.Collector.register(Collector.java:132)
	nl.nlighten.prometheus.tomcat.TomcatMetricsServlet.init(TomcatMetricsServlet.java:17)
	org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:496)
	org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81)
	org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:650)
	org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342)
	org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:803)
	org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66)
	org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:800)
	org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1463)
	org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	java.lang.Thread.run(Thread.java:748)
```

Maybe TomcatMetricsServlet could have it's own CollectorRegistry?
Can help with a PR if needed.

**Comment by nlighten** [comment]
For my understanding, are you doing a reload like in described [here](http://tomcat.apache.org/tomcat-9.0-doc/manager-howto.html#Reload_An_Existing_Application)? Is this from your IDE? 

I ask this because I haven't seen people doing this in a long time . Usually a full tomcat (or container) restart is done.

**Comment by peergynt** [comment]
@nlighten yes, this seems to happen when the metrics application is restarted (stop and then start) or reloaded via the manager.
We actually try to avoid restarting the entire container as much as possible and prefer managing applications individually whenever feasible.

Maybe a combination of using a private CollectorRegistry and clearing it first in the servlet init() method could solve this problem.

**Comment by nlighten** [comment]
Personally I would strongly discourage running multiple business applications in one Tomcat instance and/or restart them through the manager but I'll stop preaching about that.

The solution is probably not a private CollectorRegistry but but avoiding the registration of the collectors when this has already been done. The question is what you are trying achieve by restarting the metrics application. Is it to clear values or something else?

**Comment by peergynt** [comment]
In my particular case, I don't typically need to restart the metrics application.
I came across this issue on a non-production server when I modified the context.xml file and Tomcat restarted its applications.
I was just surprised that the metrics application cannot be restarted. It is not a big issue though.

I do think however that in the case of servlets, it might be best to avoid using static singleton as with the global CollectorRegistry. Hence, my suggestion to use a private field.
